extensions-core/avro-extensions/src/main/java/org/apache/druid/data/input/avro/AvroFlattenerMaker.java

```diff
@@ -128,8 +126,8 @@ public Object getRootField(final GenericRecord record, final String key)
 
   private Object transformValue(final Object field)
   {
-    if (fromPigAvroStorage && field instanceof GenericData.Array) {
-      return Lists.transform((List) field, item -> String.valueOf(((GenericRecord) item).get(0)));
+    if (field instanceof GenericData.Array) {
```

Comment:
Is this the right thing to do since the flag isn't there anymore? It seems like it should be removed to me, since the behavior without `fromPigAvroStorage` set was to skip this section, which means that it would likely get picked up by the `if (field instanceof List) { ..` block below, which does a different transformation.

Reply:
Thank you for reviewing @clintropolis, and I think you have a valid point there. I'll comment the part, for now, let us see if the unit-tests. There is a difference since the `List` below is a plain Java collection, and the `GenericData.Array` is specific, so I expect a different outcome.